functions/anonymous-functions.md

```diff
@@ -0,0 +1,198 @@
+| `const myArray = [...];`                            | Here we create an array, `myArray`.                                                                                                      |
+| `myArray.forEach(...);`                             | To iterate over each element in `myArray`, we call the `forEach` function off `myArray`.                                                 |
+| `item => { ... }`                                   | The `forEach` function takes in one argument: another function! This anonymous function is using arrow function syntax.                  |
+| `item`                                              | The anonymous function we pass into `forEach` should take in one argument itself. This argument will always represent the iterated item. |
+| `console.log(..., item);`                           | In the function body, we can access the value of `item` and read it!                                                                     |
+
+Anonymous functions can be arrow functions or non-arrow functions! Whether we use an arrow function or non-arrow function depends on the context of the problem, and which form of `this` we need.
+
+### Example: Iterating Through Tips
+
+Let's look at another example! Here, we have an array of cash tips, `cashTips`. We want to find the average cash tip. We will create a `sum` variable. Then, we'll iterate through `cashTips`. For each tip in the array, we'll increment `sum`. Then we'll calculate the average cash tip.
+
+```javascript
+const cashTips = [4, 7, 9, 12, 3, 18, 6];
+let sum = 0;
+
+cashTips.forEach((tip) => {
+  sum += tip;
+});
+
+const average = sum / cashTips.length;
+console.log(`The average cash tip is ${average}`);
+```
+
+To create the loop, we call `cashTips.forEach()`. The `forEach` function takes in one argument: another function.
+
+Here, we pass in an anonymous function:
+
+<!-- prettier-ignore-start -->
+```js
+tip => {
+  sum += tip;
+}
+```
+<!-- prettier-ignore-end -->
+
+This function takes in one argument, `tip`. Our `forEach` function will use this anonymous function, and each `tip` will be each item in `cashTips`. Inside our anonymous function, we increment `sum` by `tip`.
+
+### Anonymous Functions in Context
+
+This is a great example of an anonymous function because:
+
+- The logic of this function isn't reused anywhere
+- The function is more readable in the context of the `forEach` loop than outside it
+
+## Uses for Anonymous Functions: `map`
+
+Another great use of anonymous functions is with the `map` function.
+
+The responsibility of `map` is to look at an array, and create another array whose items are transformed, or _mapped_, from the original array.
+
+For example, imagine that we're working with an API to look at our social media users. We get back an array of user IDs, but each ID in this array is a string. We need to convert these strings into numbers in order for us to use them!
```

Comment:
adjust the language around how map processes an array
there's no intrinsic reason why we need to convert the ids to ints, so add the assumption statement to preemptively avoid that distraction

```suggestion
The responsibility of `map` is to loop through an array, and create another array whose items are transformed, or _mapped_, from the original array.

For example, imagine that we're working with an API to look at our social media users. We get back an array of user IDs, but each ID in this array is a string. Let's assume that we need to convert these strings into numbers in order for us to use them!
```